src/assets/components/TodoContainer/TodoContainer.jsx

```diff
@@ -19,33 +27,77 @@ const TodoContainer = ({ sortDirection, sortField, setSortDirection, setSortFiel
       },
     };
 
-    const url = `https://api.airtable.com/v0/${import.meta.env.VITE_AIRTABLE_BASE_ID}/${import.meta.env.VITE_TABLE_NAME}?view=Grid%20view&sort[0][field]=${sortField}&sort[0][direction]=${sortDirection}`;
+  const url = `https://api.airtable.com/v0/${import.meta.env.VITE_AIRTABLE_BASE_ID}/${import.meta.env.VITE_TABLE_NAME}?view=Grid%20view&sort[0][field]=title&sort[0][direction]=${sortDirection}&sort[1][field]=createdAt&sort[1][direction]=${sortDirection}`;
 
     try {
       const response = await fetch(url, options);
       if (!response.ok) {
         throw new Error(`Error: ${response.status}`);
       }
-
       const data = await response.json();
       const todos = data.records.map((todo) => ({
         id: todo.id,
         title: todo.fields.title,
+        createdAt: new Date(todo.fields.createdAt),
+        status: todo.fields.status|| false,
+        category: todo.fields.category || 'other',
       }));
 
-      setTodoList(todos);
-      setIsLoading(false);
+      const categorizedTodos = {
+        study: [],
+        work: [],
+        grocery: [],
+        other: [],
+      };
+      todos.forEach((todo) => {
+        categorizedTodos[todo.category].push(todo);
+      });
+  
+     
+
+       // Filter todos by status (Completed and In Progress)
+       const completedTodos = todos.filter(todo => todo.status === true);
+       const remainingTodos = todos.filter(todo => todo.status !== true);
+ 
+       // Sort the "Remaining Todos"
+       const sortedRemainingTodos = remainingTodos.sort((a, b) => {
+         if (sortField === 'time') {
+           return sortDirection === 'asc' ? a.createdAt - b.createdAt : b.createdAt - a.createdAt;
+         }
+         return sortDirection === 'asc'
+           ? a.title.localeCompare(b.title)
+           : b.title.localeCompare(a.title);
+       });
+ 
+       // Sort the "Completed Todos"
+       const sortedCompletedTodos = completedTodos.sort((a, b) => {
+         if (sortField === 'time') {
+           return sortDirection === 'asc' ? a.createdAt - b.createdAt : b.createdAt - a.createdAt;
+         }
+         return sortDirection === 'asc'
+           ? a.title.localeCompare(b.title)
+           : b.title.localeCompare(a.title);
+       });
+ 
+       // Set the state for lists
+       setTodoList(sortedRemainingTodos);
+       setCompletedTodoList(sortedCompletedTodos);
+       setTodosByCategory(categorizedTodos);
+       setIsLoading(false);
+
     } catch (error) {
       console.error('Error fetching data:', error.message);
     }
   };
 
   useEffect(() => {
     fetchData();
-  }, [sortDirection, sortField]);
+  }, [sortDirection, sortField, selectedCategory]);
 
   const addTodo = (newTodo) => {
     setTodoList((prevTodoList) => [...prevTodoList, newTodo]);
```

Comment:
why arent you using the api to add the todo list. by the time you call fetchData() the todo you added will be deleted